pkg/distributor/push_test.go

```diff
@@ -846,6 +845,114 @@ func TestHandler_ErrorTranslation(t *testing.T) {
 		})
 	}
 }
+func TestHandler_HandleRetryAfterHeader(t *testing.T) {
+	const (
+		DefaultRetryBase      = 3 * time.Second
+		DefaultMaxAllowedAtts = 2
+	)
+
+	testCases := []struct {
+		name           string
+		parserError    bool
+		handlerError   error
+		expectedStatus int
+		expectRetry    bool
+		retryCfg       *RetryConfig
+		retryAttempt   int
+		minRetryAfter  int
+		maxRetryAfter  int
+	}{
+		{
+			name:           "Request canceled, HTTP 499, no Retry-After",
+			parserError:    false,
+			handlerError:   context.Canceled,
+			expectedStatus: statusClientClosedRequest,
+			expectRetry:    false,
+			retryCfg:       &RetryConfig{Base: DefaultRetryBase, Enabled: true, MaxAllowedAttempts: DefaultMaxAllowedAtts},
```

Comment:
i'd add a test case where the config is disabled too (either with `false` or with `Base: 0` if do that change)

Reply:
it is the second case `Generic error, HTTP 500, no Retry-After`. or do you mean when error code is not retryable and we want to disable the retry?